app/src/androidTest/java/com/github/sdp/mediato/DatabaseTests/CollectionsTests.java

```diff
@@ -0,0 +1,129 @@
+package com.github.sdp.mediato.DatabaseTests;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.github.sdp.mediato.data.Database;
+import com.github.sdp.mediato.model.Location;
+import com.github.sdp.mediato.model.Review;
+import com.github.sdp.mediato.model.User;
+import com.github.sdp.mediato.model.media.Collection;
+import com.github.sdp.mediato.model.media.Media;
+import com.github.sdp.mediato.model.media.MediaType;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+
+@RunWith(AndroidJUnit4.class)
+
+/**
+ * This class contains all the tests for database collections interactions
+ */
+public class CollectionsTests {
+    private final static int STANDARD_COLLECTION_TIMEOUT = 10;
+    User user1 = new User.UserBuilder("uniqueId1")
+            .setUsername("user_test_1")
+            .setEmail("email_test_1")
+            .setRegisterDate("09/03/2023")
+            .setLocation(new Location(3.14, 3.14))
+            .build();
+
+    Collection collection1;
+    Collection collection2;
+
+    Review review1 = new Review(user1.getUsername(), new Media(MediaType.MOVIE, "Harry Potter 1", "the chosen one", "url", 1));
+    Review review2 = new Review(user1.getUsername(), new Media(MediaType.MOVIE, "Harry Potter 2", "the chosen two", "url", 2), 9);
+    Review review3 = new Review(user1.getUsername(), new Media(MediaType.MOVIE, "Harry Potter 3", "the chosen three", "url", 3), 2, "meh");
+
+    Map<String, Review> reviews1 = new HashMap<>() {
+    };
+    Map<String, Review> reviews2 = new HashMap<>() {
+    };
+    Map<String, Review> reviews3 = new HashMap<>() {
+    };

```

Comment:
Can put those variable in final and private?

Reply:
I'll put them private but not final since they're initialised in other parts of the code and their values can change.